util/util.go

```diff
@@ -210,6 +210,20 @@ func FindInInterface(iface interface{}, mapKeys ...string) (interface{}, error)
 	return mapped, nil
 }
 
+// CreateTemp Creates a temporary directory so that we may gather results and files before compressing the final
+// artifact. It returns a string representing a filepath, and an optional error.
+func CreateTemp() (string, error) {
+	tmp, err := os.MkdirTemp(os.TempDir(), "hcdiag")
```

Comment:
I think this is a fairly big change from current behavior, as we are putting the temp directory in the OS's temp directory instead of the destination directory, and we don't include the timestamp piece of the directory name. I wonder if we should keep similar behavior in this particular change. 🤔

Reply:
Good point about the agent start time. I hadn't considered that!